src/Engine/ProtoCore/FFI/CLRFFIFunctionPointer.cs

```diff
@@ -452,8 +457,8 @@ public override object Execute(ProtoCore.Runtime.Context c, Interpreter dsi)
             else
             {
                 retVal = base.Execute(c, dsi);
+                marshaller.OnDispose(s.Last(), c, dsi); //Notify marshler for dispose.
```

Comment:
Can we do something to ensure "marshaller.OnDispose()" is called before the actual dispose of thisObject, otherwise in another scenario this issue will remain unresolved.